### PR TITLE
Add client.RPCSender for sending client requests using our RPC transport.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: kv/db.go [hunk 1]
```diff
@@ -146,3 +147,78 @@ func (s *DBServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", contentType)
 	w.Write(body)
 }
+
+// RegisterRPC ...
+func (s *DBServer) RegisterRPC(rpcServer *rpc.Server) error {
+	return rpcServer.RegisterName("Server", &s.rpcDBServer)
+}
+
+type rpcDBServer struct {
```
Comment (bdarnell) [inline kv/db.go]: If this type is just to create a separate method namespace for RPCs, we should probably use the same pattern as in multiraft.go: https://github.com/cockroachdb/cockroach/blob/master/multiraft/multiraft.go#L106

Comment (petermattis) [inline kv/db.go]: Yep, just for creating a separate method namespace. I like the multiraft approach better. Changed.

File: kv/db.go [hunk 2]
```diff
@@ -146,3 +147,78 @@ func (s *DBServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", contentType)
 	w.Write(body)
 }
+
+// RegisterRPC ...
+func (s *DBServer) RegisterRPC(rpcServer *rpc.Server) error {
+	return rpcServer.RegisterName("Server", &s.rpcDBServer)
```
Comment (bdarnell) [inline kv/db.go]: The "service" namespace is a little confusing: here we register the suite of RPCs under the "Server" service, and in node.go we register the same methods under "Node".

File: server/server.go [hunk 3]
```diff
@@ -119,6 +119,7 @@ func NewServer(ctx *Context, stopper *util.Stopper) (*Server, error) {
 	s.stopper.AddCloser(s.raftTransport)
 
 	s.kvDB = kv.NewDBServer(sender)
+	s.kvDB.RegisterRPC(s.rpc)
```
Comment (bdarnell) [inline server/server.go]: We should put the server side of this behind a flag or something until we've made a more permanent decision about it.

**Comment by bdarnell** [comment]
LGTM

File: kv/db.go (hunk 2 above)
Comment (petermattis) [inline kv/db.go]: Yeah, it is confusing. Note that the "Node." endpoints differ from the "Server." endpoints due to the different senders: `Node` uses a `LocalSender` while `DBServer` uses `TxnCoordSender` wrapped around a `DistSender`.

File: server/server.go (hunk 3 above)
Comment (petermattis) [inline server/server.go]: Ok. Added `Context.ExperimentalRPCServer` which defaults to false.

**Pull request merged**